src/autocomplete/typeahead.js

```diff
@@ -562,9 +573,7 @@ function buildDom(options) {
         options.datasets[0] && options.datasets[0].displayKey ? 'both' : 'list'),
       // Indicates whether the dropdown it controls is currently expanded or collapsed
       'aria-expanded': 'false',
-      // If a placeholder is set, label this field with itself, which in this case,
-      // is an explicit pointer to use the placeholder attribute value.
-      'aria-labelledby': ($input.attr('placeholder') ? $input.attr('id') : null),
+      'aria-labelledby': ariaLabelledBy,
```

Comment:
`aria-labelledby` refers to an `id`, I'm pretty sure this should've been `aria-label`, since it's actually the text you want to be spoken (in the case of a placeholder, and in the case of adding your custom text)

[see this SO post for example](https://stackoverflow.com/questions/19616893/difference-between-aria-label-and-aria-labelledby)

Reply:
Hi @Haroenv, thanks for the link. 

`aria-labelledby` is what I need control of since it creates a problem when the placeholder differs from the label of the input. If you don't think a custom id is beneficial I could remove that, but what I really need is the ability to omit the `aria-labelledby` attribute.

Our current workaround is to remove this field everywhere we use autocomplete.js where label and placeholder differ in order to get screen readers to behave properly, so I think this change would be helpful. 

```
$.fn.removeAriaLabelledByAttr = function removeAriaLabelledByAttr() {
  return $(this).removeAttr('aria-labelledby');
};
```

Reply:
I actually think that our original code was wrong, and totally understand why you'd want `aria-labelledby` not to be present, which would also be necessary for `aria-label`

Reply:
cc @kaelig any inputs on this PR. Since you worked on this subject (aria autocomplete). Now I have trouble understanding exactly what this PR does. Can you add documentation @jrparish? Will be clearer

Reply:
@vvo @Haroenv 
I documented the usage, but I'm thinking maybe we should just remove this `aria-labelledby` field altogether and instead add an `ariaLabel` option for custom labels?

Reply:
I have no particular opinion on this matter, if someone knowledgable like you think it's the best then let's do that. Just keep in mind about breaking changes if there are, I think those aria labels are never used for styling (like input[aria..]) so should be safe.

Reply:
>I documented the usage, but I'm thinking maybe we should just remove this aria-labelledby field altogether and instead add an ariaLabel option for custom labels?

I'd say yes for that change

Reply:
The idea behind this PR is good. The self-referencing `aria-labelledby` was kind of a hack to avoid having to deal with a visually hidden label which could have broken some existing implementations. So I agree with what's been said above, `aria-label` would work great. However, if `aria-label` wasn't specified, we could keep a self referencing `aria-labelledby` (this explicitly tells screen readers to use the placeholder as a label).

So I guess you could do:

```js
// 1. If aria-label is set, keep it as-is and omit aria-labelledby
// 2. If a placeholder is set, explicitly self-reference the input
// to ensure assistive technologies pick it up as if it were the label
// 3. If none of those were specified, do nothing (and shed a tear for accessibility)
'aria-labelledby':
  ($input.attr('aria-label') ? null : ($input.attr('placeholder') ? $input.attr('id') : null)),
```

Or (same effect):

```js
// 1. If aria-label was set, use it
// 2. If no aria-label was set, try using the placeholder as a fallback
// 3. If none of those were specified, do nothing (and shed a tear for accessibility)
'aria-label':
  ($input.attr('aria-label') ? $input.attr('aria-label') : ($input.attr('placeholder') ? $input.attr('placeholder') : null)),
```